swayfmt/tests/mod.rs

```diff
@@ -1,82 +1,95 @@
 use std::sync::Arc;
 use swayfmt::{config::user_def::FieldAlignment, Formatter};
 
-/// Checks if the formatter is producing the same output when given it's output.
-fn test_stability(formatted_input: String, formatter: Formatter) -> bool {
-    let mut formatter = formatter;
-    let formatted_sway_code =
-        Formatter::format(&mut formatter, Arc::from(formatted_input.clone()), None).unwrap();
-    formatted_input == formatted_sway_code
+#[macro_use]
+mod macros;
+
+/// Takes a configured formatter as input and formats a given input and checks the actual output against an
+/// expected string. The check runs format() twice to ensure the received output does not change on a second pass.
+fn check_custom_config(unformatted: &str, expected: &str, formatter: &mut Formatter) {
```

Comment:
I was thinking if I should have a separate fn that accepts a `Formatter` for when `Config`s change, and went ahead and implemented this separate `check_custom_config`.

The idea was that when focusing in on writing a single test, you do not have to write a few lines to init a `Formatter` if you do not need any change in settings, but rather just call `check()` and then write your expected and unformatted strings there. Imo less thinking needed that way, + it's clear which tests are using custom configs by checking `check()` vs `check_custom_config()` usages. Probably could use a better name..?

Reply:
> Probably could use a better name..?

Maybe better to match what the function signature represents? E.g. `check_with_formatter` or similar?